internal/command/deploy/machines.go

```diff
@@ -312,3 +341,117 @@ func releaseLease(ctx context.Context, machine *api.Machine) error {
 
 	return nil
 }
+
+func (md *machineDeployment) setStrategy(passedInStrategy string) {
+	if passedInStrategy != "" {
+		md.strategy = passedInStrategy
+		return
+	}
+	stratFromConfig := md.appConfig.GetDeployStrategy()
+	if stratFromConfig != "" {
+		md.strategy = stratFromConfig
+		return
+	}
+	// FIXME: any other checks we want to do here? e.g., we used to do canary if any max_per_region==0 && app.distinct_regions?==false
+	md.strategy = "rolling"
+}
+
+func (md *machineDeployment) createReleaseInBackend(ctx context.Context) error {
+	_ = `# @genqlient
+	mutation MachinesCreateRelease($input:CreateReleaseInput!) {
+		createRelease(input:$input) {
+			release {
+				id
+				version
+			}
+		}
+	}
+	`
+	input := gql.CreateReleaseInput{
+		AppId:           md.appConfig.AppName,
+		Image:           md.img.Tag,
+		PlatformVersion: "machines",
+		Strategy:        gql.DeploymentStrategy(strings.ToUpper(md.strategy)),
+		Definition:      md.appConfig.Definition,
+	}
+	resp, err := gql.MachinesCreateRelease(ctx, md.gqlClient, input)
+	if err != nil {
+		return err
+	}
+	md.releaseId = resp.CreateRelease.Release.Id
+	md.releaseVersion = resp.CreateRelease.Release.Version
+	return nil
+}
+
+func (md *machineDeployment) flyRelease() string {
+	return fmt.Sprintf("%s_%d", md.releaseId, md.releaseVersion)
+}
+
+func (md *machineDeployment) baseMachineConfig() *api.MachineConfig {
+	machineConfig := &api.MachineConfig{}
+	machineConfig.Metadata = map[string]string{
+		api.MachineConfigMetadataKeyFlyPlatformVersion: api.MachineFlyPlatformVersion2,
+		api.MachineConfigMetadataKeyFlyRelease:         md.flyRelease(),
```

Comment:
consider splitting this out into two keys:

fly_release_id = md.releaseId
fly_release_version = md.releaseVersion